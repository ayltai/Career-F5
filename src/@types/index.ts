export type { Course, } from './Course';
export type { Experience, } from './Experience';
export type { Profession, } from './Profession';
export type { RouteItem, } from './RouteItem';
export type { Section, } from './Section';
export type { Segmentation, } from './Segmentation';
