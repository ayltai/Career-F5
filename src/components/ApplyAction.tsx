import { RocketLaunch as RocketLaunchIcon, } from '@mui/icons-material';
import { Button, Typography, } from '@mui/material';
import React from 'react';
import { useTranslation, } from 'react-i18next';

export const ApplyAction = ({
    onClick,
} : {
    onClick : () => void,
}) => {
    const { t, } = useTranslation();

    return (
        <Button
            color='primary'
            variant='contained'
            startIcon={
                <RocketLaunchIcon sx={{
                    color : 'white',
                }} />
            }
            onClick={onClick}>
            <Typography
                color='white'
                variant='button'>
                {t('action_apply')}
            </Typography>
        </Button>
    );
};
