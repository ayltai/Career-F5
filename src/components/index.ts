export { ApplyAction, } from './ApplyAction';
export { ApplyDialog, } from './ApplyDialog';
export { ContactAction, } from './ContactAction';
export { Footer, } from './Footer';
export { PaymentPlan, } from './PaymentPlan';
export { RangeIndicator, } from './RangeIndicator';
export { TopBar, } from './TopBar';
