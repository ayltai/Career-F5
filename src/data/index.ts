import type { Experience, } from '../@types';

export const experiences : Experience[] = [
    {
        key  : 'experience_jpmc',
        path : '/images/jpmc.webp',
    },
    {
        key  : 'experience_hsbc',
        path : '/images/hsbc.webp',
    },
    {
        key  : 'experience_mox',
        path : '/images/mox.webp',
    },
    {
        key  : 'experience_chc',
        path : '/images/chc.webp',
    },
    {
        key  : 'experience_nhs',
        path : '/images/nhs.webp',
    },
    {
        key  : 'experience_generali',
        path : '/images/generali.webp',
    },
];

export const contactDefinition = {
    properties : {
        'entry.2143157713' : {
            title     : 'form_name',
            type      : 'string',
            minLength : 3,
        },
        'entry.1731291251' : {
            title     : 'form_email',
            type      : 'string',
            minLength : 3,
            format    : 'email',
            pattern   : '^[a-z0-9!#$%&\'*+/=?^_`{|}~-]+(?:\\.[a-z0-9!#$%&\'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?$',
        },
        'entry.445180598'  : {
            title     : 'form_message',
            type      : 'string',
            minLength : 3,
            widget    : 'Text',
            view      : {
                rows : 5,
            },
        },
    },
    required   : [
        'entry.2143157713',
        'entry.1731291251',
        'entry.445180598',
    ],
};

export const applicationDefinition = {
    properties : {
        'entry.2143157713' : {
            title     : 'form_name',
            type      : 'string',
            minLength : 3,
        },
        'entry.1731291251' : {
            title     : 'form_email',
            type      : 'string',
            minLength : 3,
            format    : 'email',
            pattern   : '^[a-z0-9!#$%&\'*+/=?^_`{|}~-]+(?:\\.[a-z0-9!#$%&\'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?$',
        },
        'entry.875538261' : {
            title  : 'form_specialisation',
            type   : 'string',
            enum   : [
                'course_backend_java',
                'course_backend_node',
                'course_frontend_react',
                'course_frontend_android',
                'course_frontend_react_native',
                'course_cloud_aws',
                'course_cloud_azure',
                'course_cloud_gcp',
            ],
            widget : 'Select',
        },
        'entry.445180598' : {
            title     : 'form_message_optional',
            type      : 'string',
            widget    : 'Text',
            view      : {
                rows : 5,
            },
        },
    },
    required   : [
        'entry.2143157713',
        'entry.1731291251',
        'entry.875538261',
    ],
};
