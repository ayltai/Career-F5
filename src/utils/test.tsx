import { ThemeProvider, } from '@mui/material';
import { configureStore, PreloadedState, Store, } from '@reduxjs/toolkit';
import '@testing-library/jest-dom/extend-expect';
import { render, } from '@testing-library/react';
import mediaQuery from 'css-mediaquery';
import React, { ReactNode, } from 'react';
import { Provider, } from 'react-redux';

import { preferencesReducer, } from '../states';
import { AppTheme, } from '../styles';

export const createMatchMedia = (width : number) => (query : string) => ({
    matches             : mediaQuery.match(query, {
        width,
    }),
    media               : '',
    onchange            : null,
    addListener         : () => {},
    removeListener      : () => {},
    addEventListener    : () => {},
    removeEventListener : () => {},
    dispatchEvent       : () => false,
});

const createStore = (preloadedState? : Record<string, any>) => configureStore({
    reducer : {
        preferences : preferencesReducer,
    },
    preloadedState,
});

const defaultStore = createStore();

type RootState = ReturnType<typeof defaultStore.getState>;

const customRender = (ui : any, {
    preloadedState,
    store = createStore(preloadedState),
    ...rest
} : {
    preloadedState?  : PreloadedState<RootState>,
    store?           : Store<RootState>,
    [ rest : string] : any,
} = {}) => render(ui, {
    wrapper : ({
        children,
    } : {
        children : ReactNode,
    }) => (
        <Provider store={store}>
            <ThemeProvider theme={AppTheme('en')}>
                {children}
            </ThemeProvider>
        </Provider>
    ),
    ...rest,
});

export * from '@testing-library/react';

export { customRender as render, };
