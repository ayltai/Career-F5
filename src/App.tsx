import { WhatsApp as WhatsAppIcon, } from '@mui/icons-material';
import { Button, CssBaseline, Dialog, DialogActions, DialogContent, DialogContentText, DialogTitle, Fab, Stack, Theme, ThemeProvider, Typography, } from '@mui/material';
import mixpanel from 'mixpanel-browser';
import React, { Fragment, useEffect, useState, } from 'react';
import { useTranslation, } from 'react-i18next';

import { ApplyAction, ApplyDialog, ContactAction, Footer, TopBar, } from './components';
import { applicationDefinition, } from './data';
import { Home, } from './screens';
import { useAppSelector, } from './hooks';
import { AppTheme, } from './styles';
import { handleError, } from './utils';

const contactAction = <ContactAction />;

const handleCallToActionClick = () => {
    if (process.env.REACT_APP_WHATSAPP_PHONE_NUMBER) {
        const newWindow = window.open(`https://wa.me/${ process.env.REACT_APP_WHATSAPP_PHONE_NUMBER }`, '_blank', 'noopener,noreferrer');
        if (newWindow) newWindow.opener = null;
    }
};

export const App = () => {
    const language = useAppSelector(state => state.preferences.language);

    const [ applyOpen, setApplyOpen, ] = useState<boolean>(false);
    const [ errorOpen, setErrorOpen, ] = useState<boolean>(false);
    const [ appTheme,  setAppTheme,  ] = useState<Theme>(AppTheme(language));

    const { t, i18n, } = useTranslation();

    const handleApplyOpen = () => setApplyOpen(true);

    const handleApplyClose = () => setApplyOpen(false);

    const handleErrorClose = () => setErrorOpen(false);

    const handleSubmit = (data : Record<string, string>) => fetch('https://corsproxy.io/?https://docs.google.com/forms/u/0/d/e/1FAIpQLSf2J4dcklFWXhHhO6glY5zJa6T9zreNR5uP-j8k3Eu_m1jgzQ/formResponse', {
        method  : 'POST',
        headers : {
            'Content-Type' : 'application/x-www-form-urlencoded',
        },
        body    : new URLSearchParams(data).toString(),
    }).then(() => {
        setApplyOpen(false);

        return true;
    }).catch(e => {
        handleError(e);

        setErrorOpen(true);

        return false;
    });

    const applyAction = (
        <ApplyAction onClick={handleApplyOpen} />
    );

    useEffect(() => mixpanel.track('Home'), []);

    useEffect(() => {
        i18n.changeLanguage(language).catch(handleError);
        setAppTheme(AppTheme(language));
    }, [ language, ]);

    return (
        <Fragment>
            <CssBaseline />
            <ThemeProvider theme={appTheme}>
                <a id='home' />
                <TopBar
                    branding={
                        <Fragment>
                            <Typography
                                marginRight={2}
                                variant='h4'
                                fontFamily='"Lobster Two"'
                                fontWeight='bold'>
                                {t('branding')}
                            </Typography>
                            <img
                                width={40}
                                height={40}
                                alt={t<string>('branding')}
                                src='/safari-pinned-tab.svg' />
                        </Fragment>
                    }
                    menu={[
                        {
                            text  : t<string>('courses'),
                            route : '#courses',
                        },
                        {
                            text  : t<string>('contact'),
                            route : '#contact',
                        },
                        {
                            text  : t<string>('about'),
                            route : '#instructor',
                        },
                    ]}
                    callToAction={
                        <Stack
                            spacing={2}
                            direction='row'>
                            {applyAction}
                            {contactAction}
                        </Stack>
                    }
                />
                <Fab
                    sx={{
                        display  : {
                            sx : 'block',
                            md : 'none',
                        },
                        position : 'fixed',
                        right    : 32,
                        bottom   : 32,
                    }}
                    color='secondary'
                    onClick={handleCallToActionClick}>
                    <WhatsAppIcon sx={{
                        color : 'white',
                    }} />
                </Fab>
                <Home
                    applyAction={applyAction}
                    contactAction={contactAction} />
                <Footer />
                <ApplyDialog
                    open={applyOpen}
                    onClose={handleApplyClose}
                    title={t<string>('apply_title')}
                    description={t<string>('apply_description')}
                    definition={applicationDefinition}
                    onSubmit={handleSubmit} />
                <Dialog
                    open={errorOpen}
                    onClose={handleErrorClose}>
                    <DialogTitle>
                        {t<string>('error_title')}
                    </DialogTitle>
                    <DialogContent>
                        <DialogContentText>
                            {t<string>('error_apply_failed')}
                        </DialogContentText>
                    </DialogContent>
                    <DialogActions>
                        <Button onClick={handleErrorClose}>
                            {t<string>('action_close')}
                        </Button>
                    </DialogActions>
                </Dialog>
            </ThemeProvider>
        </Fragment>
    );
};
