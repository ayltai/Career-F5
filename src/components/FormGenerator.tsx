import { Button, Card, CardActions, CardContent, Typography, useMediaQuery, } from '@mui/material';
import { GridContainer, widgets, } from '@ui-schema/ds-material';
import { createEmptyStore, createOrderedMap, injectPluginStack, isInvalid, relT, StoreSchemaType, storeUpdater, TranslatorContext, UIMetaProvider, UIStoreProvider, } from '@ui-schema/ui-schema';
import React, { useCallback, useState, } from 'react';
import { useTranslation, } from 'react-i18next';

import { handleError, } from '../utils';

const GridStack = injectPluginStack(GridContainer);

export type FormGeneratorProps = {
    title?       : string,
    description? : string,
    definition   : object,
    onSubmit     : (data : Record<string, string>) => Promise<boolean>,
};

export const FormGenerator = ({
    elevated = true,
    title,
    description,
    definition,
    onSubmit,
} : FormGeneratorProps & {
    elevated? : boolean,
}) => {
    const storeSchema = createOrderedMap({
        type : 'object',
        ...definition,
    });

    const [ store,        setStore,        ] = useState(() => createEmptyStore(storeSchema.get('type')));
    const [ showValidity, setShowValidity, ] = useState(false);
    const [ isSubmitting, setIsSubmitting, ] = useState(false);

    const desktopMode = useMediaQuery('(min-width:900px)');

    const { t, } = useTranslation();

    const handleChange = useCallback((actions : any) => setStore(storeUpdater(actions)), [ setStore,]);

    const handleSubmit = () => {
        setShowValidity(true);

        if (!isInvalid(store.getValidity())) {
            setIsSubmitting(true);

            onSubmit(store.valuesToJS() as Record<string, string>).then(success => {
                if (success) {
                    setStore(createEmptyStore(storeSchema.get('type')));

                    setShowValidity(false);
                }
            }).then(() => setIsSubmitting(false)).catch(handleError);
        }
    };

    const translate = (text : string, context? : TranslatorContext, schema? : StoreSchemaType) => {
        const schemaT = relT(schema, context);

        return schemaT ? schemaT : t(text);
    };

    return (
        <Card
            sx={{
                borderRadius : 4,
                padding      : desktopMode ? 4 : 1,
            }}
            elevation={elevated ? 4 : 0}>
            <CardContent>
                {title && (
                    <Typography
                        gutterBottom
                        variant='h5'
                        fontWeight='bold'>
                        {t(title)}
                    </Typography>
                )}
                {description && (
                    <Typography
                        gutterBottom
                        variant='body1'>
                        {t(description)}
                    </Typography>
                )}
                <UIMetaProvider
                    t={translate}
                    widgets={widgets}>
                    <UIStoreProvider
                        showValidity={showValidity}
                        store={store}
                        onChange={handleChange}>
                        <GridStack
                            isRoot
                            schema={storeSchema} />
                    </UIStoreProvider>
                </UIMetaProvider>
            </CardContent>
            <CardActions>
                <Button
                    sx={{
                        marginX : 1,
                    }}
                    disabled={isSubmitting || !!isInvalid(store.getValidity())}
                    variant='contained'
                    onClick={handleSubmit}>
                    <Typography
                        color='white'
                        variant='button'>
                        {t(isSubmitting ? 'action_submitting' : 'action_submit')}
                    </Typography>
                </Button>
            </CardActions>
        </Card>
    );
};
