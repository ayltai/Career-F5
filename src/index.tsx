import * as Sentry from '@sentry/react';
import mixpanel from 'mixpanel-browser';
import React, { StrictMode, } from 'react';
import ReactDOM from 'react-dom/client';
import { Provider, } from 'react-redux';
import { PersistGate, } from 'redux-persist/integration/react';
import '@fontsource/kreon/400.css';
import '@fontsource/kreon/700.css';
import '@fontsource/lobster-two/400.css';
import '@fontsource/noto-sans-tc/chinese-traditional-400.css';
import '@fontsource/noto-sans-tc/chinese-traditional-700.css';
import '@fontsource/noto-serif-tc/chinese-traditional-400.css';
import '@fontsource/noto-serif-tc/chinese-traditional-700.css';
import '@fontsource/sansita/400.css';
import '@fontsource/sansita/400-italic.css';
import '@fontsource/sansita/700.css';

import PackageInfo from '../package.json';

import { App, } from './App';
import { applyI18n, } from './i18n';
import en from './i18n/en.json';
import zh from './i18n/zh.json';
import { persistor, store, } from './states';
import { handleError, } from './utils';
import './index.css';

if (process.env.REACT_APP_MIXPANEL_TOKEN) mixpanel.init(process.env.REACT_APP_MIXPANEL_TOKEN, {
    debug : process.env.NODE_ENV !== 'production',
});

if (process.env.REACT_APP_SENTRY_DSN) Sentry.init({
    dsn              : process.env.REACT_APP_SENTRY_DSN,
    environment      : process.env.NODE_ENV || 'development',
    release          : PackageInfo.version,
    tracesSampleRate : 1,
    integrations     : [
        new Sentry.BrowserTracing(),
    ],
});

applyI18n({
    language         : navigator.language.substring(0, 2),
    fallbackLanguage : 'en',
    resources        : {
        en : {
            translation : en,
        },
        zh : {
            translation : zh,
        },
    },
}).then(() => ReactDOM.createRoot(document.getElementById('root')!).render(
    <StrictMode>
        <Sentry.ErrorBoundary showDialog>
            <Provider store={store}>
                <PersistGate persistor={persistor}>
                    <App />
                </PersistGate>
            </Provider>
        </Sentry.ErrorBoundary>
    </StrictMode>
)).catch(handleError);
