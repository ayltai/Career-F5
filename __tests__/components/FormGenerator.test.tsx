import React from 'react';

import { contactDefinition, } from '../../src/data';
import { fireEvent, render, waitFor, } from '../../src/utils/test';
import { FormGenerator, } from '../../src/components/FormGenerator';

describe('FormGenerator', () => {
    it('renders correctly', () => {
        const handleSubmit = jest.fn(() => Promise.resolve(true));

        const { container, getByLabelText, getByRole, } = render(
            <FormGenerator
                title='title'
                description='description'
                definition={contactDefinition}
                onSubmit={handleSubmit} />
        );

        expect(container).toMatchSnapshot();

        expect(getByLabelText('form_name *')).toBeInTheDocument();
        expect(getByLabelText('form_email *')).toBeInTheDocument();
        expect(getByLabelText('form_message *')).toBeInTheDocument();
        expect(getByRole('button')).toBeInTheDocument();
        expect(getByRole('button')).toBeDisabled();
    });

    it('enables submit button when form is valid', async () => {
        const handleSubmit = jest.fn(() => Promise.resolve(true));

        const { getAllByRole, getByRole, } = render(
            <FormGenerator
                title='title'
                description='description'
                definition={contactDefinition}
                onSubmit={handleSubmit} />
        );

        fireEvent.change(getAllByRole('textbox')[0], {
            target : {
                value : 'name',
            },
        });

        fireEvent.change(getAllByRole('textbox')[1], {
            target : {
                value : 'email',
            },
        });

        fireEvent.change(getAllByRole('textbox')[2], {
            target : {
                value : 'message',
            },
        });

        await waitFor(() => expect(getByRole('button')).toBeDisabled());

        fireEvent.change(getAllByRole('textbox')[1], {
            target : {
                value : 'email@test.com',
            },
        });

        await waitFor(() => expect(getByRole('button')).toBeEnabled());
    });

    it('calls onSubmit when submit button is clicked', async () => {
        const handleSubmit = jest.fn(() => Promise.resolve(true));

        const { getAllByRole, getByRole, } = render(
            <FormGenerator
                title='title'
                description='description'
                definition={contactDefinition}
                onSubmit={handleSubmit} />
        );

        fireEvent.change(getAllByRole('textbox')[0], {
            target : {
                value : 'name',
            },
        });

        fireEvent.change(getAllByRole('textbox')[1], {
            target : {
                value : 'email@test.com',
            },
        });

        fireEvent.change(getAllByRole('textbox')[2], {
            target : {
                value : 'message',
            },
        });

        await waitFor(() => expect(getByRole('button')).toBeEnabled());

        fireEvent.click(getByRole('button'));

        await waitFor(() => {
            expect(getByRole('button')).toBeDisabled();
            expect(handleSubmit).toHaveBeenCalledTimes(1);
            expect(handleSubmit).toHaveBeenCalledWith({
                'entry.2143157713' : 'name',
                'entry.1731291251' : 'email@test.com',
                'entry.445180598'  : 'message',
            });
        });
    });
});
