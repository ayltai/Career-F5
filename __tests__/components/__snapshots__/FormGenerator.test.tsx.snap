// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormGenerator renders correctly 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation4 MuiCard-root css-encc62-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-1ek6cck-MuiTypography-root"
      >
        title
      </h5>
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-1fds5j6-MuiTypography-root"
      >
        description
      </p>
      <div
        class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-ge2zft-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="uis-1"
                  id="uis-1-label"
                >
                  form_name
                  <span
                    aria-hidden="true"
                    class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                  >
                     
                    *
                  </span>
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1491elt-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    id="uis-1"
                    minlength="3"
                    required=""
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-yjsfm1"
                    >
                      <span>
                        form_name
                         
                        *
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-ge2zft-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="uis-2"
                  id="uis-2-label"
                >
                  form_email
                  <span
                    aria-hidden="true"
                    class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                  >
                     
                    *
                  </span>
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1491elt-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    id="uis-2"
                    minlength="3"
                    pattern="^[a-z0-9!#$%&'*+/=?^_\`{|}~-]+(?:\\.[a-z0-9!#$%&'*+/=?^_\`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?$"
                    required=""
                    type="email"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-yjsfm1"
                    >
                      <span>
                        form_email
                         
                        *
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-ge2zft-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="uis-3"
                  id="uis-3-label"
                >
                  form_message
                  <span
                    aria-hidden="true"
                    class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                  >
                     
                    *
                  </span>
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline css-fhtslh-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <textarea
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
                    id="uis-3"
                    minlength="3"
                    required=""
                    style="height: 0px; overflow: hidden;"
                  />
                  <textarea
                    aria-hidden="true"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
                    readonly=""
                    style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding: 0px; width: 100%;"
                    tabindex="-1"
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-yjsfm1"
                    >
                      <span>
                        form_message
                         
                        *
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1cfib1r-MuiButtonBase-root-MuiButton-root"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <span
          class="MuiTypography-root MuiTypography-button css-1we6fc-MuiTypography-root"
        >
          action_submit
        </span>
      </button>
    </div>
  </div>
</div>
`;
